### PR TITLE
Implement LoanCheckout model for gathering history

### PR DESCRIPTION
**Pull request by attemoi**
## Description

Context found at https://jira.lingsoft.fi/browse/SIMPLYE-195

## How Has This Been Tested?

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/model/patron.py [hunk 1]
```diff
@@ -571,6 +576,30 @@ def until(self, default_loan_period):
...
+
+    patron_id = Column(
+        Integer, ForeignKey("patrons.id", ondelete="CASCADE"), index=True
+    )
+    patron: Patron
+
+    license_pool_id = Column(Integer, ForeignKey("licensepools.id"), index=True)
+    license_pool: Mapped[LicensePool] = relationship("LicensePool")
+
+    license_id = Column(Integer, ForeignKey("licenses.id"), index=True, nullable=True)
```
Comment (attemoi) [inline core/model/patron.py:594]: Is license_id needed?

File: tests/api/test_circulationapi.py [hunk 2]
```diff
@@ -173,6 +174,51 @@ def test_borrow_sends_analytics_event(self, circulation_api: CirculationAPIFixtu
...
+        assert 1 == len(circulation_api.patron.loan_checkouts)
+
+        # Now try to renew the book.
+        circulation_api.remote.queue_checkout(loaninfo)
+        loan, hold, is_new = self.borrow(circulation_api)
+
+        # Renewals are counted as checkouts
+        assert 2 == len(circulation_api.patron.loan_checkouts)
+
+        # Loans of open-access books go through a different code
```
[the remark this answers is not in the document]
Comment (sylvain-degeilh-lingsoft) [reply tests/api/test_circulationapi.py:214]: We don't know if there will be such titles in the library collection, at least not at the beginning.

File: core/model/patron.py (hunk 1 above)
Comment (sylvain-degeilh-lingsoft) [reply core/model/patron.py]: @ttuovinen or @jompu might be able to answer.

**Comment by attemoi** [comment]
Something to note is that the next time there are database changes in the upstream project, the changes from this downstream migration need to be merged back into the upstream branch. This process is explained here: https://alembic.sqlalchemy.org/en/latest/branches.html

**Pull request merged**